parser/parser.py

```diff
@@ -145,6 +151,66 @@ def p_statement_if(p):
     'statement : IF expression THEN statement ELSE THEN statement END_OF_IF_THEN_ELSE END_OF_STATEMENT'
     p[0] = ('if-statement', p[2], p[4], p[7])
 
+def p_variable_parameters(p):
+    'parameters : NAME LIST_ITEM_SEPARATOR parameters'
+    p[0] = ('parameters', p[1], p[3])
+
+def p_variable_parameters_base(p):
+    '''parameters : NAME
+                  | empty'''
+    p[0] = ('parameters', p[1])
+
+def p_empty(p):
+    'empty :'
+    p[0] = None
+
+'''
+----------------- functions start ----------------- bare for oversikt
+'''
+def p_function_assign(p):
+    'statement : FUNCTION NAME WITH_PARAMS parameters START_OF_FUNCTION statement RETURN expression END_OF_FUNCTION'
+    p[0] = ('assign-function', p[2], p[4], p[6], p[8])
+
+def p_function_assign_no_params_no_return(p):
+    'statement : FUNCTION NAME START_OF_FUNCTION statement END_OF_FUNCTION'
+    p[0] = ('assign-function', p[2], ('literal-expression', None), p[4], ('literal-expression', None))
+
+def p_function_assign_no_params(p):
+    'statement : FUNCTION NAME START_OF_FUNCTION statement RETURN expression END_OF_FUNCTION'
+    p[0] = ('assign-function', p[2], ('literal-expression', None), p[4], p[6])
+
+def p_function_assign_no_return(p):
+    'statement : FUNCTION NAME WITH_PARAMS parameters START_OF_FUNCTION statement END_OF_FUNCTION'
+    p[0] = ('assign-function', p[2], p[4], p[6], ('literal-expression', None))
+
+def p_function_assign_no_body(p):
+    'statement : FUNCTION NAME WITH_PARAMS parameters START_OF_FUNCTION RETURN expression END_OF_FUNCTION'
+    p[0] = ('assign-function', p[2], p[4], ('literal-expression', None), p[7])
+
+def p_function_assign_no_params_no_body(p):
+    'statement : FUNCTION NAME START_OF_FUNCTION RETURN expression END_OF_FUNCTION'
+    p[0] = ('assign-function', p[2], ('literal-expression', None), ('literal-expression', None), p[5])
+
+def p_function_run(p):
+    'statement : RUN NAME WITH_PARAMS list-body END_OF_STATEMENT'
```

Comment:
list-body and not params?

Reply:
yes, because it is for arguments, not parameters. Arguments can be expressions, and are just like list-body